flake.nix

```diff
@@ -2,128 +2,107 @@
-                  buildCommand = old.buildCommand + ''
-                    cp output.js $out
-                  '';
-                });
-              in
-              (old.preBuild or "") + ''
-                mkdir seabug-contracts-module
-                cp -r $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts/* seabug-contracts-module
-                cp ${seabug-contracts-output-js}/output.js seabug-contracts-module
-                rm -rf $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
-                cp -r seabug-contracts-module $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
-              '';
-          } // envVars);
-        in
-        pkgsFor.${system}.stdenv.mkDerivation {
-          name = "nft-marketplace";
-          dontUnpack = true;
-          dontBuild = true;
-          installPhase = ''
-            cp -r ${nft-marketplace-frontend}/lib/node_modules/nft-marketplace/build $out
-          '';
-        };
+      # TODO: upstream
+      make-seabug-contracts-output-js = system: inputs.seabug-contracts.packages.${system}.seabug-contracts.overrideAttrs (old: {
+        buildCommand = old.buildCommand + ''
+          cp output.js $out
+        '';
+      });
+
+      make-frontend = system: buildEnvVars: d2nFlake.packages.${system}.default.overrideAttrs (old: {
+        preBuild = (old.preBuild or "") + ''
+          mkdir seabug-contracts-module
+          cp -r $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts/* seabug-contracts-module
+          cp ${make-seabug-contracts-output-js system}/output.js seabug-contracts-module
+          rm -rf $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
+          cp -r seabug-contracts-module $out/lib/node_modules/nft-marketplace/node_modules/seabug-contracts
+          # avoid Error: error:0308010C:digital envelope routines::unsupported
+          export NODE_OPTIONS=--openssl-legacy-provider
+        '';
+        installPhase = ''
+          cp -r $out/lib/node_modules/nft-marketplace/build $out
+        '';
+      } // buildEnvVars);
     in
     {
-      # The following add a check that run `npm run test` (i.e. `craco test`)
-      # uncomment when some tests will be implemented
-
-      # checks = perSystem (system: {
-      #   cracoTest = d2nFlakeOutputs.packages.${system}.default.overrideAttrs (old: {
-      #     doCheck = true;
-      #     dontInstall = true;
-      #     checkPhase = ''
-      #       npm run test
-      #     '';
-      #   });
-      # });
+      checks = perSystem (system: {
+        format = inputs.styleguide.lib.${system}.mkCheck self;
+        cracoTest = self.packages.${system}.default.overrideAttrs (old: {
+          doCheck = true;
+          dontInstall = true;
+          checkPhase = ''
+            CI=true npm run test
+          '';
+        });
+      });
```

Comment:
Do you think is it better to keep an "useless" derivation instead of commenting it out? The only reason because there were these lines commented is that I've wrote them before discovering that no tests were implemented and I didn't want to delete them (maybe one day some tests will be implemented).

Reply:
Mmmm on second thought probably I agree with you, it requires zero effort by "nixers" when they'll add some tests.